news/_posts/2017-11-30-make-back-office-modules-great-again.md

```diff
@@ -0,0 +1,66 @@
+---
+layout: post
+title:  "Make modules on Back Office great again!"
+subtitle: "What is coming on 1.7.3 for module developers"
+date:   2017-11-30 09:10:11
+authors: [ mickaelandrieu ]
+icon: icon-blaster
+tags:
+ - 1.7
+ - 1.7.2.x
+---
+
+Since the release of 1.7.0, we are working hard on improving the way the PrestaShop developers can customize and improve shops for their customers.
+
+We have released a new theme system, introducing new helpers, widgets and inheritance system between templates: in the end it's now easier for everyone to make new themes with PrestaShop, specially for whom who are more designers than developers.
+
+We have started the migration of Back Office with the framework Symfony 2, first with Product and Module pages, then the Translation, Stock Management pages and we are actually migrating pages of Advanced Parameters section. We do it for so many reasons but to sum up, we want to focus as what we are the best: providing features for the e-merchants. Be able to Work on top of a professional, fully tested and documented framework is good for PrestaShop developers, Core team included.
+
+But it was not the main reason. The main reason is allowing every PrestaShop developer to use the Symfony ecosystem to make better, powerful and maintainable modules, to be able to evolve with the last features of PHP ecosystem.
+
+### Current situation of Back Office modules (pre 1.7.3)
+
+In Back Office, for the modern pages (migrated to Symfony 2), PrestaShop developers can still use Hooks like in the old ways. But they are not allowed to override templates, forms or data returned by the controllers. This is forbidden by design, because we are removing the old `override` system.
+
+In Product page, we have introduced a specific a specific hook called `hookDisplayAdminProductsExtra` to display modules in a specific tab, but it's really painful to hook into the forms to add new fields, new buttons or new blocks of information.
```

Comment:
2 thinks here:
* All modules adding stuff to the product page can do it via the hook `hookDisplayAdminProductsExtra`. This hook existed before the 1.7, and now all modules are merged into one additional tab of the product page.
* We also added many hooks for each tabs. Module can add content in the top or bottom of each tab. (See here for more details: https://github.com/PrestaShop/PrestaShop/pull/7613)

Reply:
Agree with you for the hooks, maybe I can add a capture to display all available hooks in Product Page too (easy to do for me)?

Reply:
There is one screenshot on the PR I linked (https://cloud.githubusercontent.com/assets/6768917/26400517/33b8fe72-4079-11e7-9e17-9d62b68c19ef.png). Let me know if you need something else.